### PR TITLE
Add managed mode

### PR DESCRIPTION
**Pull request by ineiti**
Adds the V2.1 specification that allows an organizer to create
multiple rooms but keep only one passphrase to protect fraudulous
notifications.

### CODE REVIEW

File: lib/src/v2/crowd_notifier_primitives.ts [hunk 1]
```diff
@@ -26,6 +28,47 @@ export function setupHA(): IKeyPair {
...
+/**
+ * Generates the static part for v2.1 from a pass phrase. The idea is to
+ * have a static part that doesn't change, to secure the notifications without
+ * having to keep too many keys.
+ * @param pkh public key of the health authority
+ * @param pp pass phrase, if possible 256 bits entropy or more
+ */
+export function genOrgStatic(pkh: Uint8Array, pp: string): IOrganizerData {
+  const mskO = new mcl.Fr();
+  mskO.setHashOf(from_string(pp));
```
Comment (wouterl) [inline lib/src/v2/crowd_notifier_primitives.ts:40]: This should probably be something a bit more secure, such as BPKDF2 or something of the sort.

File: lib/src/v2/crowd_notifier_primitives.ts [hunk 2]
```diff
@@ -26,6 +28,47 @@ export function setupHA(): IKeyPair {
...
+ */
+export function genOrgCode(org: IOrganizerData, info: Uint8Array):
+    ILocationData {
+  const nonce1 = randombytes_buf(NONCE_LENGTH);
+  const nonce2 = randombytes_buf(NONCE_LENGTH);
+
+  const pEnt = {nonce1, nonce2};
+  const mtr = {
+    mpk: org.mpk,
+    mskl: org.mskO,
```
Comment (wouterl) [inline lib/src/v2/crowd_notifier_primitives.ts:65]: why not use `mskl` instead of `msk0`?

File: lib/src/v2/structs.ts [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import mcl from 'mcl-wasm';
 
+export interface IOrganizerData {
+    mpk: mcl.G2;
+    mskO: mcl.Fr;
```
Comment (wouterl) [inline lib/src/v2/structs.ts:4]: see above, but now I see why you call it `mskO`. Works for me :).

File: lib/src/v2/crowd_notifier_primitives.ts (hunk 2 above)
Comment (wouterl) [reply lib/src/v2/crowd_notifier_primitives.ts:65]: (never mind, `mskO` (as in Organization), makes sense to me)

**Review by wouterl — COMMENTED** [review]
Looks good.

File: lib/src/v2/crowd_notifier_primitives.ts (hunk 2 above)
Comment (ineiti) [reply lib/src/v2/crowd_notifier_primitives.ts:65]: Still copying stuff from your documents - to be sure I'll understand it later and know where it comes from... ;)

File: lib/src/v2/crowd_notifier_primitives.ts (hunk 1 above)
Comment (ineiti) [reply lib/src/v2/crowd_notifier_primitives.ts:40]: There is no such npm :(

I'm currently creating a random 256-bit hex string as a pass phrase in the web UI, so I don't think BPKDF2 will change anything there. Are you OK if I just increase the disclaimer in the `genOrgStatic` method? Currently it reads:

```
 * @param pp pass phrase, if possible 256 bits entropy or more
```

So I could beef it up to something like:

```
 * @param pp pass phrase, needs to have at least 256 bit of entropy. E.g.: 
 * Buffer.from(randomBytes(32)).toString('hex')
```

**Pull request merged**